frontend/src/pages/SignupAuth.js

```diff
@@ -0,0 +1,46 @@
+import { json, redirect } from 'react-router-dom';
+
+import Signup from '../components/Authentication/Signup.js';
+
+export default function SignupAuthPage() {
+    return <Signup />;
+}
+
+
+
+export async function signupAction({ request }) {
+    const mode = 'signup';
+
+    const data = await request.formData();
+    const authData = {
+        email: data.get('email'),
+        password: data.get('password'),
+    };
+    console.log('Auth Data:', authData);
+
```

Comment:
Validate the presence of email and password in authData before making the fetch request
 if (!authData.email || !authData.password) {
    return json({ message: 'Email and password are required.' }, { status: 422 });
  }